packages/shared/src/components/fields/PasswordField.tsx

```diff
@@ -45,17 +46,20 @@ export function PasswordField({
     ? `Password need a minimum length of ${props.minLength}`
     : passwordStrengthStates[passwordStrengthLevel].label;
   const hint = !hasUserAction ? null : userActionHint;
-  const shouldShowStrength = showStrength && hasUserAction;
+  const shouldShowStrength = !!value && showStrength && hasUserAction;
   const Icon = useType === 'password' ? EyeIcon : EyeCancelIcon;
 
+  const onChange = (input: string) => {
+    setPasswordStrengthLevel(passwordStrength(input).id);
+    setValue(input);
+  };
+
   return (
     <TextField
       {...props}
       autoComplete="off"
       type={useType}
-      valueChanged={(value) =>
-        setPasswordStrengthLevel(passwordStrength(value).id)
-      }
+      valueChanged={onChange}
       leftIcon={<LockIcon />}
       hint={showStrength ? hint : props.hint}
```

Comment:
But then should this not be the shouldShowStrength comparison?

Reply:
I thought there were some ideas behind why this was left out to not use it. I agree it is better to consolidate the conditions using a single parameter. Let me refactor this.